### PR TITLE
New: py-dalib, py-qdldl, py-qpsolvers; Update: py-ecos, py-scs, py-osqp

### PR DESCRIPTION
**Pull request by meyersbs**
```mermaid
graph TD;
  'py-dalib'-->'py-qpsolvers';
  'py-qpsolvers'-->'py-ecos';
  'py-qpsolvers'-->'py-osqp';
  'py-osqp'-->'py-qdldl';
  'py-qpsolvers'-->'py-scs';
```

### CODE REVIEW

**Comment by meyersbs** [comment]
@spackbot fix style

**Review by tldahlgren — COMMENTED** [review]
Confirmed the 7 version sha256.  Deferring merge decision to @adamjstewart and or @pradyunsg

File: var/spack/repos/builtin/packages/py-osqp/package.py [hunk 1]
```diff
@@ -13,10 +13,18 @@ class PyOsqp(PythonPackage):
...
+    )
     version("0.6.1", sha256="47b17996526d6ecdf35cfaead6e3e05d34bc2ad48bcb743153cefe555ecc0e8c")
 
     depends_on("cmake", type="build")
     depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@40.8.0:", when="@0.6.2.post8:", type="build")
+    depends_on("py-setuptools-scm@6.2:", when="@0.6.2.post8:", type="build")
     depends_on("py-numpy@1.7:", type=("build", "run"))
     depends_on("py-scipy@0.13.2:", type=("build", "run"))
     depends_on("py-future", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-osqp/package.py]: Future is no longer needed

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
Love the dependency graph! I wonder if we could teach spackbot to automatically add this to PRs...

**Comment by meyersbs** [comment]
> Love the dependency graph! I wonder if we could teach spackbot to automatically add this to PRs...

The dependency graphs help keep @qwertos sane when we're working internally.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**